src/primitives.jl

```diff
@@ -183,22 +183,36 @@ dropnull(X::NullableVector) = X.values[!X.isnull]    # -> Vector
 """
     dropnull!(X::AbstractVector)
 
-Remove null entries of `X` in-place and return the updated vector. This is a
-no-op if no nulls are present.
-
-    dropnull!(X::NullableVector)
-
-Remove null entries of `X` in-place and return a `Vector` view of `X.values`
+Remove null entries of `X` in-place and return a `Vector` view of the
+unwrapped `Nullable` entries. If no nulls are present, this is a no-op
+and `X` is returned.
 """
 function dropnull!(X::AbstractVector)
     if !(Nullable <: eltype(X))
         return X
     end
     deleteat!(X, find(isnull, X))
+    res = similar(X)
+    for i in eachindex(X, res)
+        res[i] = isa(X[i], Nullable) ? X[i].value : X[i]
```

Comment:
Isn't this condition unnecessary since at the top of the function you're checking that the element type is `Nullable`?